WooCommerce/WooCommerceTests/ViewRelated/HubMenu/HubMenuViewModelTests.swift

```diff
@@ -160,6 +160,58 @@ final class HubMenuViewModelTests: XCTestCase {
         // Then
         XCTAssertEqual(viewModel.menuElements, [.woocommerceAdmin, .viewStore, .reviews])
     }
+    func test_hub_menu_returns_expected_store_and_woocommerce_admin_url() {
+        // Given
+        let sampleURL = "https://testshop.com/"
+        let sampleAdminURL = "https://testshop.com/wp-admin/"
+        let sessionManager = SessionManager.testingInstance
+        let site = Site.fake().copy(url: sampleURL,
+                                    adminURL: sampleAdminURL)
+        sessionManager.defaultSite = site
+        let stores = MockStoresManager(sessionManager: sessionManager)
+
+        // When
+        let viewModel = HubMenuViewModel(siteID: site.siteID,
+                                         stores: stores)
+
+        // Then
+        XCTAssertEqual(viewModel.storeURL, try sampleURL.asURL())
+        XCTAssertEqual(viewModel.woocommerceAdminURL, try sampleAdminURL.asURL())
+    }
+    func test_hub_menu_returns_default_woocommerce_url_if_store_url_is_empty() {
+        // Given
+        let sampleURL = ""
+        let sampleAdminURL = ""
+        let sessionManager = SessionManager.testingInstance
+        let site = Site.fake().copy(url: sampleURL,
+                                    adminURL: sampleAdminURL)
+        sessionManager.defaultSite = site
+        let stores = MockStoresManager(sessionManager: sessionManager)
+
+        // When
+        let viewModel = HubMenuViewModel(siteID: site.siteID,
+                                         stores: stores)
+        // Then
+        XCTAssertEqual(viewModel.storeURL, WooConstants.URLs.blog.asURL())
+        XCTAssertEqual(viewModel.woocommerceAdminURL, WooConstants.URLs.blog.asURL())
+    }
+    func test_that_store_and_admin_url_do_not_return_nil_if_store_url_is_nil() {
```

Comment:
Happy to get some feedback on how to better name these tests 😅

Reply:
I think of names like this:
`test_methodName_expected_result_condition`
e.g.

`test_storeURL_is_not_nil_when_site_has_nil_url`

Personally I think it looks better when it's snake case between the parts, and camel case for the parts, but that isn't the agreed convention: it should all be snake case apart from method names and other quoted code.

Reply:
Implicit in the above: I would separate out the tests for storeURL and woocommerceAdminURL

Reply:
I would add `when` and `then` also in the test function,

func test_<property or method name, no camel case>_when<test condition>_then_<expected result>

e.g

`func test_ woocommerceAdminURL_when_site_adminURL_is_empty_then_returns_fallback`

Reply:
That is very helpful, thanks!